node/core/pvf/src/prepare/worker.rs

```diff
@@ -342,11 +357,26 @@ async fn recv_response(stream: &mut UnixStream, pid: u32) -> io::Result<PrepareR
 ///
 ///	7. Send the result of preparation back to the host. If any error occurred in the above steps, we
 ///	   send that in the `PrepareResult`.
-pub fn worker_entrypoint(socket_path: &str) {
+pub fn worker_entrypoint(socket_path: &str, node_version: Option<&str>) {
 	worker_event_loop("prepare", socket_path, |rt_handle, mut stream| async move {
+		let worker_pid = std::process::id();
+		let mut version_checked = false;
 		loop {
-			let worker_pid = std::process::id();
 			let (pvf, dest) = recv_request(&mut stream).await?;
+			if !version_checked {
```

Comment:
Why isn't that done outside of the loop?

Reply:
Node<->worker exchange is half-duplex, to send an error response, we must accept a request first. But yes, it's invariant to the loop body, so I should have placed that outside the loop and only fired it once after the first request. I'll fix it.